react/facilitator_controls.js

```diff
@@ -107,6 +122,15 @@ class FacilitatorControls extends Component {
     await fetch('/api/fill-seats/'); 
   }
 
+  setSelectionError() {
+    this.setState({selectionError: true });
+  }
+
+  setSelectionFalse() {
+    this.setState({selectionError: false });
+  }
```

Comment:
any particular reason we have two functions for this?  seems like it would be easiest to have one that takes a boolean input that we could feed to the call to `setState`

Reply:
We definitely tried that, and I'm not exactly sure why but it caused an infinite loop. It was really bizarre.

Reply:
Can you show me what you tried?

Reply:
We made the setSelection function look like this 
```
setSelectionError(arg) {
    this.setState({selectionError: arg });
  }
```
and then the component looks like this
```
<Button variant="primary" onClick={this.state.selectedTeam1Id && this.state.selectedTeam2Id ? this.setBuggedEnvelopes : this.setSelectionError(true)}>
                Submit Changes
              </Button>
              <Modal show={this.state.selectionError} onHide={this.setSelectionError(false)}>

```
This is the error: 

<img width="1180" alt="Screen Shot 2021-01-27 at 8 27 46 AM" src="https://user-images.githubusercontent.com/34659460/106021396-9ca06a00-6079-11eb-8a2f-3474cc6dd039.png">

Reply:
right - so the difference here is that in your current changes, you're passing function references to `onHide`, which is to say "execute this function when the modal is hidden".

in the code you just posted, what you're doing instead is evaluating `this.setSelectionError` with the argument `false`, so when `render` is called, this function immediately evaluates.  the infinite loop is happening because every call to `setState` causes a rerender, so what you're seeing is this:

```
render -> this.setSelectionError(false) -> this.setState -> render -> this.setSelectionError(false) -> etc
```

so your problem is that you weren't passing a function reference to `onHide`, you were evaluating a function, then passing its result (`undefined`) to `onHide`.